### PR TITLE
fix: GDD Send Amount Input Field

### LINKED ISSUE #287: 🐛 [Bug] Number separator not behaving properly
## 🐛 Bugreport

When sending GDD the user has to put in a number including a decimal divider. We have a backend part which converts this and a frontend part wich handles this aswell. While i can send GDD with `.` and  `,` in English, I cannot in German.
When sending `.` in English it is the correct behaviour. While `,` is converted by the backend which seems inconsistent.

Furthermore it is possible to input letters and spaces into the numberfield aswell. This should be prevented.

### LINKED ISSUE #404: 🐛 [Bug] Send GDD number input is not valid for less then two trailing digits
## 🐛 Bugreport

Send GDD number input is not valid for less then two digits after comma. Therefore values like `450,5` or `450` are not valid. You need to have two digits after the comma like `450,50` and '450,00`. This is inconvenient for the user.

Have the number field add the missing trailing digits actively?

### LINKED ISSUE #405: 🐛 [Bug] Send GDD number can scroll in negative
## 🐛 Bugreport

Send GDD number can scroll in negative

By pressing the arrows on the send-gdd gdd-input-field you can scroll into negative values. Even tho you get a passive validation hint that this is not a valid value, this should be prevented.

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest
 * added vue-focus package as global directive
 * GDD send amount input field is now a text field
 * the field has its own validation rule with locale
 
### Issues
- fixes #405 
- fixes #404 
- fixes #287

### CODE REVIEW

File: frontend/src/views/Pages/AccountOverview/GddSend/TransactionConfirmation.vue [hunk 1]
```diff
@@ -1,44 +1,46 @@
...
-        </b-col>
-        <b-col class="text-right">
-          <b-button variant="success" :disabled="loading" @click="$emit('send-transaction')">
-            {{ $t('form.send_now') }}
-          </b-button>
-        </b-col>
-      </b-row>
-    </b-col>
-  </b-row>
+  <div>
```
Comment (Mogge) [inline frontend/src/views/Pages/AccountOverview/GddSend/TransactionConfirmation.vue:2]: I added a div on top level. So we can have two rows inside the div. One for the information and one for the buttons. Before it was a row nested into a row.

File: frontend/src/views/Pages/AccountOverview/GddSend/TransactionConfirmation.vue [hunk 2]
```diff
@@ -1,44 +1,46 @@
...
+    <b-row>
+      <b-col>
+        <div class="display-4 p-4">{{ $t('form.send_check') }}</div>
+        <b-list-group>
+          <b-list-group-item class="d-flex justify-content-between align-items-center">
+            {{ email }}
+            <b-badge variant="primary" pill>{{ $t('form.receiver') }}</b-badge>
+          </b-list-group-item>
+          <b-list-group-item class="d-flex justify-content-between align-items-center">
+            {{ $n(amount, 'decimal') }} GDD
```
Comment (Mogge) [inline frontend/src/views/Pages/AccountOverview/GddSend/TransactionConfirmation.vue:12]: This line changed.

File: frontend/src/views/Pages/AccountOverview/GddSend/TransactionConfirmation.vue [hunk 3]
```diff
@@ -1,44 +1,46 @@
...
+            <b-badge variant="primary" pill>{{ $t('form.message') }}</b-badge>
+          </b-list-group-item>
+          <b-list-group-item class="d-flex justify-content-between align-items-center">
+            {{ $d($moment(date), 'long') }}
+            <b-badge variant="primary" pill>{{ $t('form.date') }}</b-badge>
+          </b-list-group-item>
+        </b-list-group>
+      </b-col>
+    </b-row>
+    <b-row class="mt-4">
```
Comment (Mogge) [inline frontend/src/views/Pages/AccountOverview/GddSend/TransactionConfirmation.vue:26]: added some margin top here

File: frontend/src/views/Pages/AccountOverview/GddSend/TransactionForm.vue [hunk 4]
```diff
@@ -53,10 +53,9 @@
                 :name="$t('form.amount')"
                 :rules="{
                   required: true,
-                  double: [2, $i18n.locale === 'de' ? ',' : '.'],
-                  between: [0.01, balance],
+                  gddSendAmount: [0.01, balance],
                 }"
-                v-slot="{ errors }"
+                v-slot="{ errors, valid }"
```
Comment (Mogge) [inline frontend/src/views/Pages/AccountOverview/GddSend/TransactionForm.vue:58]: the valid slot is a boolean

**Review by Mogge — COMMENTED** [review]
Please test

**Review by ogerly — APPROVED** [review]

**Pull request merged**